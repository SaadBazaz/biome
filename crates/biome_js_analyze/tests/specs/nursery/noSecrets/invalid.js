const apiKey = '12345';
const password = 'mySecretPassword';
var secret = "AKIA1234567890EXAMPLE"