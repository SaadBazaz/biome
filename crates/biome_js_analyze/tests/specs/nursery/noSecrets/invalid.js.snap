---
source: crates/biome_js_analyze/tests/spec_tests.rs
expression: invalid.js
---
# Input
```jsx
const awsApiKey = "AKIA1234567890EXAMPLE"
const slackToken = "xoxb-123456789012-123456789012-abcdefghijklmnopqrstuvwx";
const rsaPrivateKey = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA1234567890..."
const facebookToken = "facebook_app_id_12345abcde67890fghij12345";
const twitterApiKey = "twitter_api_key_1234567890abcdefghijklmnopqrstuvwxyz";
const githubToken = "github_pat_1234567890abcdefghijklmnopqrstuvwxyz";
const clientSecret = "abcdefghijklmnopqrstuvwxyz"
const herokuApiKey = "heroku_api_key_1234abcd-1234-1234-1234-1234abcd5678";
const genericSecret = "secret_1234567890abcdefghijklmnopqrstuvwxyz";
const genericApiKey = "api_key_1234567890abcdefghijklmnopqrstuvwxyz";
const slackKey = "https://hooks.slack.com/services/T12345678/B12345678/abcdefghijklmnopqrstuvwx"
const twilioApiKey = "SK1234567890abcdefghijklmnopqrstuv";
const dbUrl = "postgres://user:password123@example.com:5432/dbname";
```

# Diagnostics
```
invalid.js:1:19 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
  > 1 │ const awsApiKey = "AKIA1234567890EXAMPLE"
      │                   ^^^^^^^^^^^^^^^^^^^^^^^
    2 │ const slackToken = "xoxb-123456789012-123456789012-abcdefghijklmnopqrstuvwx";
    3 │ const rsaPrivateKey = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA1234567890..."
  
  i AWS API Key
  

```

```
invalid.js:2:20 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    1 │ const awsApiKey = "AKIA1234567890EXAMPLE"
  > 2 │ const slackToken = "xoxb-123456789012-123456789012-abcdefghijklmnopqrstuvwx";
      │                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    3 │ const rsaPrivateKey = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA1234567890..."
    4 │ const facebookToken = "facebook_app_id_12345abcde67890fghij12345";
  
  i High entropy value detected
  

```

```
invalid.js:3:23 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    1 │ const awsApiKey = "AKIA1234567890EXAMPLE"
    2 │ const slackToken = "xoxb-123456789012-123456789012-abcdefghijklmnopqrstuvwx";
  > 3 │ const rsaPrivateKey = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA1234567890..."
      │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    4 │ const facebookToken = "facebook_app_id_12345abcde67890fghij12345";
    5 │ const twitterApiKey = "twitter_api_key_1234567890abcdefghijklmnopqrstuvwxyz";
  
  i RSA Private Key
  

```

```
invalid.js:4:23 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    2 │ const slackToken = "xoxb-123456789012-123456789012-abcdefghijklmnopqrstuvwx";
    3 │ const rsaPrivateKey = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA1234567890..."
  > 4 │ const facebookToken = "facebook_app_id_12345abcde67890fghij12345";
      │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    5 │ const twitterApiKey = "twitter_api_key_1234567890abcdefghijklmnopqrstuvwxyz";
    6 │ const githubToken = "github_pat_1234567890abcdefghijklmnopqrstuvwxyz";
  
  i High entropy value detected
  

```

```
invalid.js:5:23 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    3 │ const rsaPrivateKey = "-----BEGIN RSA PRIVATE KEY-----\nMIIEpAIBAAKCAQEA1234567890..."
    4 │ const facebookToken = "facebook_app_id_12345abcde67890fghij12345";
  > 5 │ const twitterApiKey = "twitter_api_key_1234567890abcdefghijklmnopqrstuvwxyz";
      │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    6 │ const githubToken = "github_pat_1234567890abcdefghijklmnopqrstuvwxyz";
    7 │ const clientSecret = "abcdefghijklmnopqrstuvwxyz"
  
  i High entropy value detected
  

```

```
invalid.js:6:21 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    4 │ const facebookToken = "facebook_app_id_12345abcde67890fghij12345";
    5 │ const twitterApiKey = "twitter_api_key_1234567890abcdefghijklmnopqrstuvwxyz";
  > 6 │ const githubToken = "github_pat_1234567890abcdefghijklmnopqrstuvwxyz";
      │                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    7 │ const clientSecret = "abcdefghijklmnopqrstuvwxyz"
    8 │ const herokuApiKey = "heroku_api_key_1234abcd-1234-1234-1234-1234abcd5678";
  
  i High entropy value detected
  

```

```
invalid.js:7:22 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    5 │ const twitterApiKey = "twitter_api_key_1234567890abcdefghijklmnopqrstuvwxyz";
    6 │ const githubToken = "github_pat_1234567890abcdefghijklmnopqrstuvwxyz";
  > 7 │ const clientSecret = "abcdefghijklmnopqrstuvwxyz"
      │                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    8 │ const herokuApiKey = "heroku_api_key_1234abcd-1234-1234-1234-1234abcd5678";
    9 │ const genericSecret = "secret_1234567890abcdefghijklmnopqrstuvwxyz";
  
  i High entropy value detected
  

```

```
invalid.js:9:23 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
     7 │ const clientSecret = "abcdefghijklmnopqrstuvwxyz"
     8 │ const herokuApiKey = "heroku_api_key_1234abcd-1234-1234-1234-1234abcd5678";
   > 9 │ const genericSecret = "secret_1234567890abcdefghijklmnopqrstuvwxyz";
       │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    10 │ const genericApiKey = "api_key_1234567890abcdefghijklmnopqrstuvwxyz";
    11 │ const slackKey = "https://hooks.slack.com/services/T12345678/B12345678/abcdefghijklmnopqrstuvwx"
  
  i High entropy value detected
  

```

```
invalid.js:10:23 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
     8 │ const herokuApiKey = "heroku_api_key_1234abcd-1234-1234-1234-1234abcd5678";
     9 │ const genericSecret = "secret_1234567890abcdefghijklmnopqrstuvwxyz";
  > 10 │ const genericApiKey = "api_key_1234567890abcdefghijklmnopqrstuvwxyz";
       │                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    11 │ const slackKey = "https://hooks.slack.com/services/T12345678/B12345678/abcdefghijklmnopqrstuvwx"
    12 │ const twilioApiKey = "SK1234567890abcdefghijklmnopqrstuv";
  
  i High entropy value detected
  

```

```
invalid.js:11:18 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
     9 │ const genericSecret = "secret_1234567890abcdefghijklmnopqrstuvwxyz";
    10 │ const genericApiKey = "api_key_1234567890abcdefghijklmnopqrstuvwxyz";
  > 11 │ const slackKey = "https://hooks.slack.com/services/T12345678/B12345678/abcdefghijklmnopqrstuvwx"
       │                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    12 │ const twilioApiKey = "SK1234567890abcdefghijklmnopqrstuv";
    13 │ const dbUrl = "postgres://user:password123@example.com:5432/dbname";
  
  i Slack Webhook
  

```

```
invalid.js:12:22 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    10 │ const genericApiKey = "api_key_1234567890abcdefghijklmnopqrstuvwxyz";
    11 │ const slackKey = "https://hooks.slack.com/services/T12345678/B12345678/abcdefghijklmnopqrstuvwx"
  > 12 │ const twilioApiKey = "SK1234567890abcdefghijklmnopqrstuv";
       │                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    13 │ const dbUrl = "postgres://user:password123@example.com:5432/dbname";
  
  i Twilio API Key
  

```

```
invalid.js:13:15 lint/nursery/noSecrets ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Potential secret found.
  
    11 │ const slackKey = "https://hooks.slack.com/services/T12345678/B12345678/abcdefghijklmnopqrstuvwx"
    12 │ const twilioApiKey = "SK1234567890abcdefghijklmnopqrstuv";
  > 13 │ const dbUrl = "postgres://user:password123@example.com:5432/dbname";
       │               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  
  i Password in URL
  

```
